src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java

```diff
@@ -127,13 +127,45 @@ private void authenticateUserOrApplication(String tokenPayload) {
     throw new ForbiddenException("Bad Token");
   }
 
-  private void authenticateApplication(String token) {
-    val application = applicationService.findByBasicToken(token);
+  protected void authenticateApplication(String tokenString) {
+    // An application just sends us Basic authToken (an id+a secret) to authenticate itself.
+    val token = BasicAuthToken.decode(tokenString);
 
-    // Deny access if they don't have a valid app token for
-    // one of our applications
-    if (application == null) {
+    if (token.isEmpty()) {
+      log.warn(
+          "AuthenticateApplication: Invalid token '"
+              + tokenString
+              + "' for application authentication request");
+      SecurityContextHolder.clearContext();
+      return;
+    }
+
+    val clientId = token.get().getClientId();
+    val clientSecret = token.get().getClientSecret();
+
+    // Deny access if they don't have a valid clientId from one of our applications
+    val application = applicationService.findByClientId(clientId);
+
+    if (application == null || application.isEmpty()) {
       SecurityContextHolder.clearContext();
+      log.warn(
+          "AuthenticateApplication: No application found for clientId '"
+              + clientId
+              + "' from token '"
+              + tokenString
+              + "'");
+      return;
+    }
+
+    // Deny access if the clientSecret in the token is wrong
+    if (!application.get().getClientSecret().equals(clientSecret)) {
+      SecurityContextHolder.clearContext();
+      log.warn(
+          "AuthenticateApplication: Wrong client secret for clientId '"
+              + clientId
+              + "' from token '"
+              + tokenString
```

Comment:
lets avoid putting the whole token string into logs as it contains the secret.

Reply:
If you feel strongly we should be logging the token for debugging purposes then maybe add a separate log at DEBUG level.